### ISSUE TITLE
False Positive: target-size input fully obscured by button is not skipped

### ISSUE THREAD

**Issue by chutchins25** [dequelabs/axe-core#5223]
#### Expectation
A form control that is fully obscured by another element (here, an `<input>` completely covered by a `<button>`) should be skipped/passed by the `target-size` rule — a fully-covered control isn't an independently actionable touch target, so it shouldn't be flagged for insufficient size.

#### Actual
The obscured `<input>` is flagged by `target-size` as a violation (false positive).

#### How to Reproduce
An `<input>` fully obscured by a `<button>` is reported by `target-size` instead of being skipped.

#### Minimal reproduction
```html
<div style="position: relative; width: 80px; height: 40px;">
  <!-- transform gives the input its own stacking context -->
  <input type="text" style="position: absolute; top: 12px; left: 32px;
         width: 16px; height: 16px; border: 0; padding: 0; transform: translateZ(0);">
  <!-- button paints on top and receives the clicks -->
  <button style="position: absolute; inset: 0; width: 80px; height: 40px;">Go</button>
</div>
```
Running `target-size` on this flags the `<input>` as *"Target has insufficient size (16px by 16px, should be at least 24px by 24px)"*, even though the button fully covers it and `document.elementFromPoint` at the input's center returns the `<button>` (the input is not clickable). Removing the `transform` (or any `opacity`/`filter`/`will-change` that creates a stacking context on the input) makes axe correctly report *"Control is ignored because it is fully obscured."* `opacity: 0.99` on the input reproduces it identically.

#### Likely cause
The "fully obscured → skip" path in `lib/checks/mobile/target-size-evaluate.js` (`filterByElmsOverlap`) relies on `hasVisualOverlap(target, neighbor)` → `visuallySort()` in `lib/commons/dom/visually-sort.js`. When the target establishes its own stacking context, `visuallySort` ranks it above the covering button, so the button isn't recognized as obscuring and the target is flagged. This is one FP of this class; the reported page may hit the same stacking-context path or a related gate (fixed/non-fixed neighbor exclusion in `findNearbyElms`, or the target rect not being fully enclosed).

**Comment by chutchins25** [comment]
Verified that #5214 (merged) fixes this.

Although #5214 is scoped to `color-contrast`, its stacking-context fix lives in `lib/commons/dom/create-grid.js`, which builds the `_stackingOrder` that `target-size` depends on via `hasVisualOverlap` → `visuallySort`. The old code ranked positioned elements (`POSITION_LEVEL = 0.3`) above real stacking contexts (`DEFAULT_LEVEL = 0.1`), so an input that established its own stacking context (`transform`/`opacity`) sorted above the covering button and wasn't recognized as obscured. #5214 sorts real and pseudo stacks at the same layer, so the button is correctly seen on top.

I confirmed this empirically by running the minimal repro from this issue through the `target-size` check in a browser, on the commit just before #5214 vs. the merged PR head:

| Repro variant | Before #5214 | After #5214 |
|---|---|---|
| input w/ `transform: translateZ(0)` | flagged (16×16 violation) — FP | `obscured` ✅ |
| input w/ `opacity: 0.99` | flagged (16×16 violation) — FP | `obscured` ✅ |
| control (no own stacking context) | `obscured` ✅ | `obscured` ✅ |

Since #5214 didn't reference this issue in its `Closes:` footer, closing manually.

**Comment by padmavemulapati** [comment]
Valdiated with the latest axe-core develop branch code base,

to reproduce/address the issues:

`target-size` should skip/pass a control that is fully obscured by another element (message: "Control is ignored because it is fully obscured and thus not clickable"). The bug was that when the obscured <input> created its own stacking context (via transform, opacity, filter, or will-change), visuallySort ranked it above the covering <button>, so axe didn't recognize it as obscured and flagged it as a 16×16 violation — a false positive.

Validations (expected results after the fix):

The fully-obscured inputs — including the ones with their own stacking context (transform: translateZ(0), opacity: 0.99) — must all be skipped/passed with the obscured message. The plain control with no stacking context (the case that always worked) is the regression guard and must also pass. As genuine violations for contrast, a small standalone target with nothing covering it should be flagged (minSize), and a partially-obscured target should give the partiallyObscured message.

https://github.com/user-attachments/assets/89bb2768-da35-4d52-8e0c-ec5ced90199a
